packages/squiggle-lang/src/js/index.ts

```diff
@@ -75,147 +60,234 @@ export function resultMap<a, b, c>(
+  return resultMap(result, x => createTsExport(x, si));
+}
+
+
+function createTsExport(x: expressionValue, sampEnv: samplingParams): squiggleExpression {
+  switch (x.tag) {
+    case "EvArray":
+      return tag("array", x.value.map(x => createTsExport(x, sampEnv)));
+    case "EvBool":
+      return tag("boolean", x.value);
+    case "EvDistribution":
+      return tag("distribution", new Distribution(x.value, sampEnv));
+    case "EvNumber":
+      return tag("number", x.value);
+    case "EvRecord":
+      return tag("record", _.mapValues(x.value, x => createTsExport(x, sampEnv)))
+  }
+}
+
+
 export function resultExn<a, c>(r: result<a, c>): a | c {
   return r.value;
 }
 
+export type point = { x: number, y: number}
+
+export type shape = {
+  continuous: point[]
+  discrete: point[]
+}
+
+function shapePoints(x : continuousShape | discreteShape): point[]{
+  let xs = x.xyShape.xs;
+  let ys = x.xyShape.ys;
+  return _.zipWith(xs, ys, (x, y) => ({x, y}))
+}
+
 export class Distribution {
   t: genericDist;
-  env: env;
+  env: samplingParams;
 
-  constructor(t: genericDist, env: env) {
+  constructor(t: genericDist, env: samplingParams) {
     this.t = t;
     this.env = env;
     return this;
   }
 
-  mapResultDist(r: result<genericDist, error>): result<Distribution, error> {
+  mapResultDist(r: result<genericDist, distributionError>): result<Distribution, distributionError> {
     return resultMap(r, (v: genericDist) => new Distribution(v, this.env));
   }
 
-  mean(): result<number, error> {
+  mean(): result<number, distributionError> {
     return Constructors_mean({ env: this.env }, this.t);
   }
 
-  sample(): result<number, error> {
+  sample(): result<number, distributionError> {
     return Constructors_sample({ env: this.env }, this.t);
   }
 
-  pdf(n: number): result<number, error> {
+  pdf(n: number): result<number, distributionError> {
     return Constructors_pdf({ env: this.env }, this.t, n);
   }
 
-  cdf(n: number): result<number, error> {
+  cdf(n: number): result<number, distributionError> {
     return Constructors_cdf({ env: this.env }, this.t, n);
   }
 
-  inv(n: number): result<number, error> {
+  inv(n: number): result<number, distributionError> {
     return Constructors_inv({ env: this.env }, this.t, n);
   }
 
-  normalize(): result<Distribution, error> {
+  normalize(): result<Distribution, distributionError> {
     return this.mapResultDist(
       Constructors_normalize({ env: this.env }, this.t)
     );
   }
 
-  toPointSet(): result<Distribution, error> {
+  shape() : result<shape, distributionError> {
+    let pointSet = toPointSet(this.t, {xyPointLength: this.env.xyPointLength, sampleCount: this.env.sampleCount}, null);
+    if(pointSet.tag === "Ok"){
+      let distribution = pointSet.value;
+      if(distribution.tag === "Continuous"){
+        return Ok({
+          continuous: shapePoints(distribution.value),
+          discrete: []
+        })
```

Comment:
I feel like we should have some class for pointSet, then that class should abstract this functionality.